src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise((resolve, reject) => {
+  const logo = document.querySelector('.logo');
+
+  if (logo) {
+    logo.addEventListener('click', () => {
+      resolve();
+    });
+  } else {
+    reject(new Error('Logo not found on the page'));
+  }
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('error'));
+  }, 3000);
+});
```

Comment:
It would be more efficient to use the `reject` function directly in the `setTimeout` call. This will make the code cleaner and easier to read.